db/dcrpg/pgblockchain.go

```diff
@@ -571,37 +626,91 @@ func (pgb *ChainDB) GetTicketPoolByDateAndInterval(maturityBlock int64,
 	return retrieveTicketsByDate(pgb.db, maturityBlock, int64(val))
 }
 
-// TicketPoolVisualization fetches the following ticketpool data: tickets
+// TicketPoolVisualization helps block consecutive access to the ticketPoolVisualization
+// method till the first request is resolved.
+func (pgb *ChainDB) TicketPoolVisualization(interval dbtypes.ChartGrouping) ([]*dbtypes.PoolTicketsData, *dbtypes.PoolTicketsData, uint64, error) {
+	// Attempt to retrieve data for the current block from cache.
+	heightSeen := pgb.Height()
+	barcharts, donutCharts, height, ok := TicketPoolData(interval, heightSeen)
+	if ok {
+		// The cache was fresh.
+		return barcharts, donutCharts, height, nil
+	}
+
+	// Cache is stale. Attempt to gain updater status.
+	if !pgb.tpUpdatePermission.TryLock() {
```

Comment:
Is this a blocking operation? Like will the process wait till the trylock is true?

Reply:
It does not block.  It just says whether or not it got the lock.